tartufo/scanner.py

```diff
@@ -348,25 +349,36 @@ def config_data(self, data: MutableMapping[str, Any]) -> None:
         self._config_data = data
 
     @cached_property
-    def excluded_findings(self) -> tuple:
-        configured_signatures = []
-        signatures = self.config_data.get("exclude_signatures", None)
-        if signatures:
-            warnings.warn(
-                "--exclude-signatures has been deprecated and will be removed in a future version. "
-                "Make sure all the exclusions are moved to exclude-findings section with new format. Example: "
-                "exclude-findings = [{signature='signature', reason='The reason of excluding the signature'}]",
-                DeprecationWarning,
-            )
-            configured_signatures.extend(signatures)
-        findings = self.config_data.get("exclude_findings", None)
-        if findings:
-            configured_signatures.extend([finding["signature"] for finding in findings])
-
-        self._excluded_findings = tuple(
-            set(self.global_options.exclude_signatures + tuple(configured_signatures))
-        )
-        return self._excluded_findings
+    def excluded_signatures(self) -> tuple:
+        if len(self._excluded_signatures) == 0:
+            signatures = list(self.global_options.exclude_signatures or ())
+            exclude_signatures = list(self.config_data.get("exclude_signatures", ()))
+            try:
+                signatures = signatures + [
+                    x for x in exclude_signatures if x not in signatures
+                ]
+                signatures = [
+                    signature["signature"]
+                    for signature in cast(List[Dict[str, str]], signatures)
+                ]
+                self._excluded_signatures = tuple(cast(List[str], signatures))
+                return self._excluded_signatures
+            except TypeError:
+                warnings.warn(
+                    "Configuring exclude-signatures as string has been deprecated and support for this format will "
+                    "be removed in the future. Please make sure to update your exclude-signatures configuration to "
+                    "an array of tables. For example: exclude-signatures = [{signature='signature', reason='The "
+                    "reason of excluding the signature'}]",
+                    DeprecationWarning,
+                )
+            try:
+                signatures = list(set(signatures + exclude_signatures))
+                self._excluded_signatures = tuple(cast(List[str], signatures))
```

Comment:
And again, we've already merged the lists and eliminated duplicates, so this can be simplified

```suggestion
                self._excluded_signatures = tuple(signatures)
```